src/org/parosproxy/paros/view/FindDialog.java

```diff
@@ -87,16 +101,25 @@ public FindDialog(Frame arg0, boolean arg1) throws HeadlessException {
      * @param parent the parent window of the FindDialog.
      * @param modal whether or not this FindDialog should be modal
      * @throws HeadlessException
+     * @see #getDialog(Window, boolean)
+     * @since TODO add version
      */
     public FindDialog(Window parent, boolean modal) throws HeadlessException {
     	super(parent, modal);
-    	initialize();
+
+    	initialize(parent);
     }
 
 	/**
 	 * This method initializes this
 	 */
-	private void initialize() {
+	private void initialize(Object parent) {
+		if (getParentsList().containsKey(parent)) {
+			this.discard();
+			getParentsList().get(parent).requestFocus();

```

Comment:
Yes, that's expected because the `FindDialog` instances are now being reused. This statement needs to be moved to the `getDialog` methods.